relay/README.md

```diff
@@ -171,31 +163,38 @@ It then waits for a response in the form of:
 <uvarint error code><uvarint msglength><message>
 ```
 
-Once it receives that, it checks if the status code is `OK`. If it is, it passes the new connection to its `NewConnHandler`.
-Otherwise, it returns the error message to the caller.
+Once it receives that, it checks if the status code is `OK`. If it is, it passes the new connection to its `NewConnHandler`. Otherwise, it returns the error message to the caller.
 
-### 'hop' protocol handler
+Peer R receives a new stream on the '/libp2p/relay/hop' protocol.
 
-Peer R receives a new stream on the 'hop' protocol.
 It then calls `readLpMaddr` twice. The first value is `<src>` and the second is `<dst>`.
-Peer R checks to make sure that `<src>` matches the remote peer of the stream its reading
-from. If it does not match, it (writes an error back?) closes the stream and halts the relay attempt.
+
+Peer R checks to make sure that `<src>` matches the remote peer of the stream its reading from. If it does not match, it writes an Error message, closes the stream and halts the relay attempt.
 
 Peer R checks if `<dst>` refers to itself, if it does, it (writes an error back?) closes the stream and halts the relay attempt.
+
 Peer R then checks if it has an open connection to the peer specified by `<dst>`.
 If it does not, and the relay is not an "active" relay it (writes an error back) closes the stream, and halts the relay attempt.
+
 If R does not have a connection to `<dst>`, and it *is* an "active" relay, it attempts to connect to `<dst>`.
+
 If this connection succeeds it continues, otherwise it (writes back an error) closes the stream, and halts the relay attempt.
-R now opens a new stream to B with the 'stop' relay protocol ID, and writes:
+
+
+#### Phase II
+
+R now opens a new stream to B with the '/libp2p/relay/stop' relay protocol multicodec and writes:
+
 ```
 <uvarint len(maddrA)><madddrA><uvarint len(maddrB)><madddrB>
 ```
 
-After this, R simply pipes the stream from A and the stream it opened to B together. R's job is complete.
+After this, R pipes the stream from A and the stream it opened to B together. R also writes OK back to A. R's job is complete.
```

Comment:
Maybe worth noting that R's job actually continues: it now needs to keep the connections alive